Automattic/wp-calypso
# Purchases: Use formatCurrency for receipts

## Proposed Changes

Receipt pages use the pre-formatted totals returned by the `/me/billing-history` endpoint. In order to make sure that their formatting matches the formatting in the rest of calypso, this PR changes them to use the `formatCurrency()` function instead.

Requires D106947-code to add some new properties to the receipts.

## Screenshots

In these screenshots I used a french locale with USD, so the prices should be displayed like `96.45 $US`.

Before:

<img width="964" alt="before-receipt-list" src="https://user-images.githubusercontent.com/2036909/229660061-fd1b6ce5-bb1b-49fc-b1e7-173deb80ac60.png">

<img width="967" alt="before-receipt" src="https://user-images.githubusercontent.com/2036909/229660065-1a3044d3-046c-4bd8-a324-94b18aa6585f.png">

After:

<img width="960" alt="after-receipt-list" src="https://user-images.githubusercontent.com/2036909/229660062-46983479-cd23-4758-873a-dfe0f12fe5eb.png">

<img width="960" alt="after-receipt" src="https://user-images.githubusercontent.com/2036909/229660063-9f9b0bdf-caf4-43cf-b86f-d0dfd3338728.png">


## Testing Instructions

- Sandbox the API and apply D106947-code.
- Use an account that has some purchases in sandbox mode. 
- (Optional: Set your account's interface language to one with right-aligned prices like French so it's easier to see the effects.)
- Visit `me/purchases/billing`.
- Verify that the prices are rendered as expected.
- Click to view an individual receipt.
- Verify that the prices are rendered as expected.
